kernel/dev/pci.h

```diff
@@ -28,6 +28,21 @@ struct pci_bar {
     bool is_mmio;
 };
 
+struct pci_driver {
+    const char *name;
+    int match;
+
+    void (*init)(struct pci_device *);
+
+    uint8_t pci_class, subclass, prog_if;
+    uint16_t vendor, device;
+};
+
+#define PCI_MATCH_CLASS 1
+#define PCI_MATCH_SUBCLASS 2
+#define PCI_MATCH_PROG_IF 4
+#define PCI_MATCH_DEVICE 8
```

Comment:
personally, i would use a shift instead of constants, if you decide to change it don't forget to wrap them in parenthesis :')